torch/ao/quantization/fx/prepare.py

```diff
@@ -437,10 +447,37 @@ def _get_target_activation_dtype_for_node(
                 and weight_dtype == torch.float16
                 and (not input_act_is_dynamic)
             ) else torch.float
+
+        weight_index = None
+        if isinstance(node, Node) and node.op == "call_function" and \
+           node.target in backend_config._pattern_complex_format_to_config:
+            weight_index = backend_config._pattern_complex_format_to_config[node.target]._input_type_to_index.get("weight")
+            if weight_index is None or weight_index >= len(node.args):
+                weight_index = None
+
+        bias_index = None
+        if isinstance(node, Node) and node.op == "call_function" and \
+           node.target in backend_config._pattern_complex_format_to_config:
+            bias_index = backend_config._pattern_complex_format_to_config[node.target]._input_type_to_index.get("bias")
+            if bias_index is None or bias_index >= len(node.args):
+                bias_index = None
+
+        input_obs_or_fq_ctr_for_args = [qconfig.activation, qconfig.activation, qconfig.activation]
+
+        WEIGHT_OBS_OR_FQ_CTR = qconfig.weight if node.target not in NON_QUANTIZABLE_WEIGHT_OPS else None
+        if weight_index is not None:
+            input_obs_or_fq_ctr_for_args[weight_index] = qconfig.weight
+
+
+        BIAS_OBS_OR_FQ_CTR = PlaceholderObserver.with_args(dtype=bias_dtype)
+        if bias_index is not None:
+            input_obs_or_fq_ctr_for_args[bias_index] = BIAS_OBS_OR_FQ_CTR
+
         return {
-            "input_act_obs_or_fq_ctr": qconfig.activation,
-            "weight_obs_or_fq_ctr": qconfig.weight,
-            "bias_obs_or_fq_ctr": PlaceholderObserver.with_args(dtype=bias_dtype),
+            "input_obs_or_fq_ctr_for_args": tuple(input_obs_or_fq_ctr_for_args),
+            # this is kept for fx graph mode quantization use cases, not supposed
+            # to be used for quantize_pt2e
+            "_input_obs_or_fq_ctr_for_kwargs": {"weight": qconfig.weight, "bias": BIAS_OBS_OR_FQ_CTR},
```

Comment:
Same here. Which nodes have such kwargs?

Reply:
Ok I see your comment, so it is useful somewhere? Curious where

Reply:
here is the test that breaks without this: (https://github.com/pytorch/pytorch/blob/master/test/quantization/fx/test_quantize_fx.py#L7331)

Reply:
is it because of weight/bias on layer_norm?

Reply:
> here is the test that breaks without this: (https://github.com/pytorch/pytorch/blob/master/test/quantization/fx/test_quantize_fx.py#L7331)

@kimishpatel here is the test the need kwargs config